The following patch to the 2.10.3 version of batch.c causes the utility to
ignore blank lines in the batch file. This is used by the leafbatch script as
a crude but effective mechanism to prevent generation of many tiny batches.

*** batch.c.orig	Wed Apr 17 10:34:04 1985
--- batch.c	Tue Aug 12 14:30:13 1986
***************
*** 87,92
  		cp = index(fname, '\n');
  		if (cp)
  			*cp = '\0';
  		nfd = fopen(fname, "r");
  		if (nfd == NULL) {
  			perror(fname);

--- 87,94 -----
  		cp = index(fname, '\n');
  		if (cp)
  			*cp = '\0';
+ 		if (fname[0] == '\0')
+ 			continue;
  		nfd = fopen(fname, "r");
  		if (nfd == NULL) {
  			perror(fname);
***************
*** 93,98
  			continue;
  		}
  		(void) fstat(fileno(nfd), &sbuf);
  		printf("#! rnews %ld\n", (long)sbuf.st_size);
  		n = 0;
  		while ((c = getc(nfd)) != EOF) {

--- 95,102 -----
  			continue;
  		}
  		(void) fstat(fileno(nfd), &sbuf);
+ 		if (sbuf.st_size == 0L)
+ 			continue;
  		printf("#! rnews %ld\n", (long)sbuf.st_size);
  		n = 0;
  		while ((c = getc(nfd)) != EOF) {
